pulsar/internal/batch_builder.go

```diff
@@ -164,7 +167,11 @@ func (bc *batchContainer) hasSpace(payload []byte) bool {
 		return true
 	}
 	msgSize := uint32(len(payload))
-	return bc.numMessages+1 <= bc.maxMessages && bc.buffer.ReadableBytes()+msgSize <= uint32(bc.maxBatchSize)
+	expectedSize := bc.buffer.ReadableBytes() + msgSize
```

Comment:
L171 compared the `bc.buffer.ReadableBytes() + msgSize` with `bc.maxMessageSize` and the original code does not.

It used to make sure that one batch size does not exceed the `maxMessageSize`. It's a part of correctly calculation whether the message is too large.

By the way, the compare code is too long to be inline.

Reply:
Good catch, but using inline looks like more clear.

Reply:
@Gleiphir2769

Reply:
@nodece This PR introduces the `maxMessageSize`. I think these changes should be related to this PR.

Reply:
> Good catch, but using inline looks like more clear, so like:

I have tried but this line is too long. It can not pass the golangci-lint

> Good catch, but using inline looks like more clear, so like:
> 
> ```
> return bc.numMessages+1 <= bc.maxMessages && expectedSize <= uint32(bc.maxBatchSize) && expectedSize <= bc.maxMessageSize
> ```

Done, thx.